.github/workflows/CI.yml

```
@@ -39,25 +48,31 @@ jobs:
             arch: x64
             num_threads: 1
             coverage: true
+
     steps:
-      - uses: actions/checkout@v2
-      - uses: julia-actions/setup-julia@v1
+      - uses: actions/checkout@v4
+
+      - uses: julia-actions/setup-julia@v2
         with:
           version: ${{ matrix.version }}
           arch: ${{ matrix.arch }}
```

This should be updated as well since currently we test `x64` on Apple Silicon runners. There are multiple options here, the main question I guess is whether we want to keep testing Julia 32bit. If not, the easiest fix is just to remove `arch` and `matrix.arch` completely. Alternatively, we could use a default of `matrix.arch: ''` and change this line to
```suggestion
          arch: ${{ (matrix.arch == '') && runner.arch || matrix.arch }}
```
which means that by default the Julia version that matches the runner's architecture is used (and hence aarch64 on Apple Silicon) but if x86 is requested explicitly it would still be used. See e.g. https://discourse.julialang.org/t/how-to-fix-github-actions-ci-failures-with-julia-1-6-or-1-7-on-macos-latest-and-macos-14/117019

---

Ooh good spot! we changed that on Turing and DPPL but I forgot here.